TombEngine/Objects/TR2/Entity/tr2_worker_flamethrower.cpp

```diff
@@ -175,35 +187,35 @@ namespace TEN::Entities::Creatures::TR2
 				}
 
 				if (Targetable(item, &AI))
-					item->Animation.TargetState = 5;
+					item->Animation.TargetState = WORKER_FLAME_STATE_ATTACK;
 				else
 				{
 					if (creature->Mood == MoodType::Attack)
-						item->Animation.TargetState = 1;
+						item->Animation.TargetState = WORKER_FLAME_STATE_IDLE;
 					else if (!AI.ahead)
-						item->Animation.TargetState = 1;
+						item->Animation.TargetState = WORKER_FLAME_STATE_IDLE;
 				}
 
 				break;
 
-			case 5:
-			case 6:
+			case WORKER_FLAME_STATE_ATTACK:
+			case WORKER_FLAME_STATE_WALK_FORWARD_ATTACK:
 				if (AI.ahead)
 				{
 					extraTorsoRot.x = AI.xAngle;
 					extraTorsoRot.y = AI.angle;
 				}
 
-				if (item->Animation.TargetState != 1 &&
-					(creature->Mood == MoodType::Escape || AI.distance > pow(SECTOR(10), 2) || !Targetable(item, &AI)))
+				if (item->Animation.TargetState != WORKER_FLAME_STATE_IDLE &&
```

Comment:
How did range check worked before? Or it was not existent at all and incorrect attack range was an original bug?